### PR TITLE
Easy plugin development

### PR DESCRIPTION
**Pull request by domanchi**
Addresses https://github.com/Yelp/detect-secrets/issues/154, by reducing the number of files that need to be changed when adding a plugin from 9 ([ref](https://github.com/Yelp/detect-secrets/pull/239/files)) to 3.

This also accidentally addresses https://github.com/Yelp/detect-secrets/issues/146, because while modifying tests, the new tests broke on this bug (which is what tests are for).

### CODE REVIEW

File: detect_secrets/core/usage.py [hunk 1]
```diff
@@ -290,74 +290,46 @@ def __new__(cls, related_args=None, **kwargs):
...
+        )
+
+    @staticmethod
+    def get_disabled_help_text(plugin):
+        for line in plugin.__doc__.splitlines():
+            line = line.strip().lstrip()
+            if line:
+                break
+        else:
+            raise NotImplementedError('Plugins must declare a docstring.')
```
Comment (KevinHock) [inline detect_secrets/core/usage.py]: ❤️

File: detect_secrets/plugins/common/util.py [hunk 2]
```diff
@@ -3,31 +3,55 @@
...
-        try:
-            if issubclass(value, BasePlugin) and value != BasePlugin:
-                mapping[value.secret_type] = key
-        except TypeError:
-            pass
-
-    return mapping
+    return {
+        value.secret_type: key
+        for key, value in import_plugins().items()
```
Comment (KevinHock) [inline detect_secrets/plugins/common/util.py]: Super nit: I know it says plugin classname above, but I liked when you did e.g.
```
for name, plugin in import_plugins().items()
```
in other places, as it was more immediately clear. / Consistency nit.

File: tests/main_test.py [hunk 3]
```diff
@@ -10,12 +10,68 @@
...
 from detect_secrets.core import audit as audit_module
 from detect_secrets.main import main
+from detect_secrets.plugins.common.util import import_plugins
 from testing.factories import secrets_collection_factory
 from testing.mocks import Any
 from testing.mocks import mock_printer
 from testing.util import uncolor
 
 
+def get_list_of_plugins(include=None, exclude=None):
```
Comment (KevinHock) [inline tests/main_test.py]: Super nit: One might expect `include` to mean 'only include', maybe `customized` and `excluded` might be more intuitive. Alternatively maybe a docstring saying `include` is only necessary for plugins w/out the defaults.

File: tests/main_test.py [hunk 4]
```diff
@@ -10,12 +10,68 @@
...
+
+
+def get_plugin_report(extra=None):
+    """
+    :type extra: Dict[str, str]
+    """
+    if not extra:
+        extra = {}
+
+    longest_name_length = max([
```
Comment (KevinHock) [inline tests/main_test.py]: Nice ➕

File: detect_secrets/plugins/base.py [hunk 5]
```diff
@@ -33,15 +60,31 @@ def __init__(self, exclude_lines_regex=None, should_verify=False, **kwargs):
...
 
         self.should_verify = should_verify
 
+    @classproperty
+    def disable_flag_text(cls):
+        name = cls.__name__
+        if name.endswith('Detector'):
+            name = name[:-len('Detector')]
+
+        # turn camel case into hyphenated strings
```
Comment (KevinHock) [inline detect_secrets/plugins/base.py]: Super consistency nit: All our other comments start cap'd

File: tox.ini [hunk 6]
```diff
@@ -14,7 +14,7 @@ commands =
     coverage run -m pytest tests -v
     coverage report --show-missing --include=tests/* --fail-under 100
     # This is so that we do not regress unintentionally
-    coverage report --show-missing --include=detect_secrets/* --omit=detect_secrets/core/audit.py,detect_secrets/core/secrets_collection.py,detect_secrets/main.py,detect_secrets/plugins/common/ini_file_parser.py --fail-under 100
+    coverage report --show-missing --include=detect_secrets/* --omit=detect_secrets/core/audit.py,detect_secrets/core/secrets_collection.py,detect_secrets/main.py,detect_secrets/plugins/common/ini_file_parser.py --fail-under 99
```
Comment (KevinHock) [inline tox.ini]: I am not a 💯 fanatic, and okay with regressing coverage. My only issue with not separating out the file(s) is that, now when someone makes a PR, including this one, it's not that clear what isn't covered. `pragma: no coverage` is another option, some people feel it's "lying" (I don't) but it definitely helps with the above.

As an aside, past code like https://github.com/Yelp/detect-secrets/commit/de7fbd265942b1d404a3dc7292db386312a9b63d#diff-895e31d74e6f4c86ab33d77ef2ff05ed  and https://github.com/Yelp/detect-secrets/commit/20d19212de53ae6175cfb2b4aebfd68fe0c18af8#diff-f56910fb7243ad4ed59cd5cf2e1af814 are semi-decent examples of lack of test coverage making things unclear as to why they were written.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -78,23 +78,19 @@ There are many examples of existing plugins to reference, under
...
-4. Update documentation
+3. Update documentation
 
    Be sure to add your changes to the `README.md` and `CHANGELOG.md` so that
    it will be easier for maintainers to bump the version and for other
    downstream consumers to get the latest information about plugins available.
 
+### Tips
+
+- There should be a total of three modified files in a minimal new plugin: the
```
Comment (KevinHock) [inline CONTRIBUTING.md]: Would `it's corresponding test,` be equivalent to what you mean?

Comment (domanchi) [reply CONTRIBUTING.md]: ah, yeah. whoops.

**Pull request merged**